### ISSUE TITLE
problems with segmented cylinder

### ISSUE THREAD

**Issue by cesare-corrado** [CGAL/cgal#93]
I create a .inr file representing the segmentation (uint8) of  cylinder and gave as input to cgal. If I define a region on the centre of the cylinder characterized by pixels having a different value (say 2) it works;. if i create the same file either with extrema regions with different pixel value, or an homogeneous cylinder, cgal is not able to create a mesh (the output file .mesh has only 9 points inside, no triangle and no tetrahedra). I think there is something not working inside the code. I can provide the input file not working, if necessary.

**Comment by lrineau** [comment]
Hi, your report is not good enough so that we can understand what the problem is. Can you give us at least:
  - an hint of the code you use (maybe copy-pasted in [Gist](http://gist.github.com/) to avoid a long listing here,
  - your operating system, and compiler, and their versions,
  - the version of CGAL you are currently using,
  - maybe other information that can help us...
Thanks.

**Comment by cesare-corrado** [comment]
I clicked on Gist and added all the information It should be in 
https://gist.github.com/cesare-corrado/55a142355655cee81581. (I'm not skilled in Gist)

**Comment by lrineau** [comment]
Thanks. Most of us do not have access to Matlab. Can you post somewhere the .inr file that "works" and the one that does not?

[WeTransfer](https://www.wetransfer.com/), [Google Drive](http://drive.google.com), [Dropbox](http://www.dropbox.com/), for example, allow to attach big files and then share them publicly with a link.

**Comment by cesare-corrado** [comment]
Yes I can try with WeTransfer, to which addres should I send? notifications@github.com is ok?

**Comment by lrineau** [comment]
You can send the mail to your own address, and then copy-paste here the URL that allows to get the files.

**Comment by cesare-corrado** [comment]
Done, the link should be:
http://we.tl/fVtAn2ydnE 
I named the files:
pipeNotWorking.inr    the case not working
pipeWorking.inr          the case working

**Comment by lrineau** [comment]
Sorry @cesare-corrado. I forgot this bug.

I can reproduce the bug. It seems the initialization of Mesh_3 does not work with `pipeNotWorking.inr`.

**Comment by lrineau** [comment]
Clément (@cjamin), with `CGAL_MESH_3_VERBOSE`, the log with the file `pipeNotWorking.inr` is:
```
construct initial points:
12/12 initial point(s) found...
Start surface scan...construct initial points:
1000/1000 initial point(s) found...
Scanning triangulation for bad facets (sequential) - number of finite facets = 60...
Number of bad facets: 4
end scan. [Bad facets:4]

Refining Surface...
Legend of the following line: (#vertices,#steps,#facets to refine,#tets to refine)
(13,0,0,0) (0.0 vertices/s)
Total refining surface time: 0.000603s

Start volume scan...Scanning triangulation for bad cells (sequential)... 28 cells scanned, done.
Number of bad cells: 0
end scan. [Bad tets:0]

Refining...
Legend of the following line: (#vertices,#steps,#facets to refine,#tets to refine)
(13,0,0,0)
Total refining volume time: 0s
Total refining time: 0.012608s

```

With `pipeWorking.inr`
```
construct initial points:
12/12 initial point(s) found...
Start surface scan...construct initial points:
1000/1000 initial point(s) found...
Scanning triangulation for bad facets (sequential) - number of finite facets = 50...
Number of bad facets: 8
end scan. [Bad facets:8]

Refining Surface...
Legend of the following line: (#vertices,#steps,#facets to refine,#tets to refine)
(40806,40793,0,0) (4435.9 vertices/s)))))
Total refining surface time: 9.19604s

Start volume scan...Scanning triangulation for bad cells (sequential)... 263464 cells scanned, done.
Number of bad cells: 140146
end scan. [Bad tets:140146]

Refining...
Legend of the following line: (#vertices,#steps,#facets to refine,#tets to refine)
(169278,138345,0,0) (8172.0 vertices/s))))))
Total refining volume time: 16.9292s
Total refining time: 26.4495s
```

I do not understand the part "`1000/1000 initial point(s) found...`".

**Comment by lrineau** [comment]
The WeTransfer link has expired. [Here is a new link, on my personnal Dropbox account](https://www.dropbox.com/s/whi2qgh5vesuvzc/pipeWorking-or-not.zip?dl=0) (for @cjamin).

**Comment by cjamin** [comment]
"1000/1000 initial point(s) found..." is there for estimating the bounding box. These points are not inserted into the mesh.

It seems that 12 initial points is not enough to start the meshing process. If I change it to 100, it works.
@lrineau: any idea why?

**Comment by lrineau** [comment]
I do not know why 12. All values are equally bad. A good strategy has to be able to recompute new initial points in case the number of initial points is not sufficient.

**Comment by cjamin** [comment]
Which is not the case for now...
Maybe we could allow the user to specify the number of points... I don't think it's possible right now, if one doesn't provide its own oracle.

**Comment by lrineau** [comment]
The case of `pipeNotWorking.inr`, with the chosen parameters, is rather strange:
  - 12 initial points are inserted,
  - the initial c3t3 must be non-empty, because a 13th point is inserted,
  - then the recomputed c3t3 from those 13 vertices, and it is empty.

**Comment by sgiraudot** [comment]
Is this bug still in CGAL or can we close the issue?

**Comment by lrineau** [comment]
The bug is still there. It is related to #294, #923.
